Orange/data/variable.py

```diff
@@ -288,95 +84,135 @@ def __init__(self, name="", compute_value=None):
         self._colors = None
 
     def make_proxy(self):
-        """
-        Copy the variable and set the master to `self.master` or to `self`.
+        """Copy the variable and set the master to `self.master` or to `self`.
 
-        :return: copy of self
-        :rtype: Variable
+        Returns
+        -------
+        Variable
+            A copy of self.
         """
         var = self.__class__()
         var.__dict__.update(self.__dict__)
         var.master = self.master
         return var
 
     def __eq__(self, other):
-        """Two variables are equivalent if the originate from the same master"""
-        return hasattr(other, "master") and self.master is other.master
+        """
+        If comparing two variables, compare masters if at least one master
+        is set (otherwise compare names).  When comparing strings, compare names,
+        otherwise, they are not equal.
+        """
+        if isinstance(other, Variable):
+            return self.master is other.master and self.master is not None
+        else:
+            return self.name == other
```

Comment:
The else clause is used for comparing with a string. By python definition ([last section paragraph](https://docs.python.org/3/reference/datamodel.html#object.__eq__)), comparing `Variable` and `str` will always use the more specific class' method.